metadata-ingestion/docs/sources/dataplex/dataplex_pre.md

```diff
@@ -0,0 +1,401 @@
+| `dataplex.entities.list` | Allows a user to view and list all entities in a zone |
+
+**Optional permissions for lineage extraction:**
+
+If you want to extract lineage information, you'll also need:
+
+| Permission                 | Description                               |
+| -------------------------- | ----------------------------------------- |
+| `datalineage.links.get`    | Allows a user to view lineage links       |
+| `datalineage.links.search` | Allows a user to search for lineage links |
+
+Default GCP Role for lineage: [roles/datalineage.viewer](https://docs.cloud.google.com/iam/docs/roles-permissions/datalineage#datalineage.viewer)
+
+#### Create a service account and assign roles
+
+1. Setup a ServiceAccount as per [GCP docs](https://cloud.google.com/iam/docs/creating-managing-service-accounts#iam-service-accounts-create-console) and assign the previously mentioned roles to this service account.
+
+2. Download a service account JSON keyfile.
+
+   Example credential file:
+
+   ```json
+   {
+     "type": "service_account",
+     "project_id": "project-id-1234567",
+     "private_key_id": "d0121d0000882411234e11166c6aaa23ed5d74e0",
+     "private_key": "-----BEGIN PRIVATE KEY-----\nMIIyourkey\n-----END PRIVATE KEY-----",
+     "client_email": "test@suppproject-id-1234567.iam.gserviceaccount.com",
+     "client_id": "113545814931671546333",
+     "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+     "token_uri": "https://oauth2.googleapis.com/token",
+     "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+     "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/test%suppproject-id-1234567.iam.gserviceaccount.com"
+   }
+   ```
+
+3. To provide credentials to the source, you can either:
+
+   Set an environment variable:
+
+   ```sh
+   $ export GOOGLE_APPLICATION_CREDENTIALS="/path/to/keyfile.json"
+   ```
+
+   _or_
+
+   Set credential config in your source based on the credential json file. For example:
+
+   ```yml
+   credential:
+     project_id: "project-id-1234567"
+     private_key_id: "d0121d0000882411234e11166c6aaa23ed5d74e0"
+     private_key: "-----BEGIN PRIVATE KEY-----\nMIIyourkey\n-----END PRIVATE KEY-----\n"
+     client_email: "test@suppproject-id-1234567.iam.gserviceaccount.com"
+     client_id: "123456678890"
+   ```
+
+### Integration Details
+
+The Dataplex connector extracts metadata from Google Dataplex using two complementary APIs:
+
+1. **Universal Catalog Entries API** (Primary, default enabled): Extracts entries from system-managed entry groups like `@bigquery`, `@pubsub`, etc. This is the recommended approach for discovering BigQuery tables and other Google Cloud resources.
+
+2. **Lakes/Zones Entities API** (Optional, default disabled): Extracts entities from Dataplex lakes and zones. Use this if you need entity-level details not available in the Entries API.
```

Comment:
wondering: what's the long term vision in google about this duality? should all be eventually served from catalog entries API?

does enabling lake/zones entities api require additional permissions?

what are those entity-level details not available in the entries api?

Reply:
I believe Google intends to get away from the Lake/Zone catalog and move towards the Universal Catalog leveraging the entries API. This is why I made it the default API for the connector but I didn't want to exclude the option of folks using the Lakes/Zone model to get that information as well.

I've updated the permissions needed for both APIs to be explicit as they are different.

Changed the wording to be explicit about entity level details to be lake/zone information.

Changes in https://github.com/datahub-project/datahub/pull/15379/commits/93be42453e19d8dc072f1bfa7b0edc548085f162.